addons/account/models/account_move_send.py

```diff
@@ -599,7 +599,7 @@ def _check_sending_data(self, moves, **custom_settings):
         assert custom_settings['pdf_report'].is_invoice_report if custom_settings.get('pdf_report') else True
         assert all(
             sending_method in dict(self.env['res.partner']._fields['invoice_sending_method'].selection)
-            for sending_method in custom_settings['sending_methods']
+            for sending_method in custom_settings.get('sending_methods') or []
```

Comment:
```suggestion
            for sending_method in custom_settings.get('sending_methods', [])
```

Reply:
If someone happens to call `_generate_blablabla` with `{'sending_method': False}` in the dict this is gonna fail, we should probably stay defensive ?

Reply:
@clbr-odoo I could agree with you but, imo, calling a method without respecting the typing of parameters is terrorism. What about calling the method with `{'sending_method': "turlututu_tsoin-tsoin"}` then? :smirk:

Reply:
`{'sending_methods': False/[]/['value']}` should all be handled imo